### DISCUSSION TITLE
LmHandler: Is it part of the API or merely a common approach for the example apps?

### DISCUSSION THREAD

**Discussion by dmeehan1968** [Lora-net/LoRaMac-node#1365] [Q&A]
LmHandler seems to provide useful functionality (Compliance testing, NVM management) but doesn't appear to be part of the LoRaMac API (due to its location in src/apps).

It appears not to be extensible, despite offering a reasonable API for handling different payload types (this appears only to be down the the hardcoded PKG_MAX_NUMBER in LmhPackage.h).  Any thoughts on the direction for this?

**Comment by janakj** [comment]
I personally find the API in src/apps hard to understand and use. And as you correctly note, it's not very extensible. In my implementation, I switched from the LmHandler API to the lower-level MIB API and never looked back. It takes more work, but it is, in my opinion, worth it in the long term.

The code in src/apps is still very useful as an example though. I am in favor of keeping it in the repo, but separately from the simpler MIB API.

**Reply by dmeehan1968** [reply]
I'd recently tried it as previously hadn't been doing NVM store/restore, a problem it seemed to solve. The rest looks like a good approach but in reality, for my apps, not actually useful. 

I'm in the process of working on a static library build of LoRaMac (PR coming) and was trying to understand if this was something to be included, but it just appears (at present) to be common code to the example apps.

**Reply by janakj** [reply]
My experience is similar to yours. I find the src/apps code valuable, but would not necessarily include it in a MAC-only static library.
